docs/001-ci-strategy.adoc

```diff
@@ -0,0 +1,101 @@
+= ADR 001 - Unit and Integration Testing CI Strategy
+Philipp Hoenisch <philipp@10101.finance>
+3.0, July 29, 2022: AsciiDoc article template
+:toc:
+:icons: font
+:attributes: 2023-01-17
+
+This document outlines the reasoning behind our CI strategy.
+
+== Decision
+
+* We have 2 stages of tests
+** Unit tests:
+In unit tests we stub all DB/io/networking/etc to achieve a sub *10 minute* runtime.
+Unit tests can run on every commit.
+** Integration tests:
```

Comment:
I guess this better applies to system tests - we we are actively testing the integration between multiple systems and their non-functional requirements, like performance, security, etc.

Reply:
You are right, I was actually talking about system tests in this context.